src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -272,7 +272,10 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 				if (imageCalloutResult.embedImage) {
 					let base64String = await this.getFileContentBase64(URI.file(imageCalloutResult.imagePath));
 					let mimeType = await this.getFileMimeType(URI.file(imageCalloutResult.imagePath));
-					this.cellModel.addAttachment(mimeType, base64String, path.basename(imageCalloutResult.imagePath).replace(' ', ''));
+					let imageName = this.cellModel.addAttachment(mimeType, base64String, path.basename(imageCalloutResult.imagePath).replace(/\s/g, ''));
+					if (path.basename(imageCalloutResult.imagePath).replace(/\s/g, '') !== imageName) {
```

Comment:
The whitespace replacement for the basename is repeated twice here. You can probably cache it in a variable to save some cycles.